src/frontend/hooks/server/invites.ts

```diff
@@ -21,23 +24,30 @@ export function usePendingInvites() {
 export function useAcceptInvite(projectId?: string) {
   const mapeoApi = useApi();
   const queryClient = useQueryClient();
-  const switchActiveProject = useUpdateActiveProjectId();
+  const switchActiveProject = usePersistedProjectId(
+    state => state.setProjectId,
+  );
 
   return useMutation({
     mutationFn: async ({inviteId}: {inviteId: string}) => {
       if (!inviteId) return;
-      mapeoApi.invite.accept({inviteId});
+      return mapeoApi.invite.accept({inviteId});
```

Comment:
it's important to actually return this value here. otherwise a couple of things will happen:

1. You won't get the return value of `invite.accept()` (doesn't currently break anything)
2. You'll potentially run into race conditions because you may try to do work after the mutation call occurs under the assumption that `invite.accept()` has resolved. However, without this change, the accept resolves on the next iteration in the event loop, breaking that assumption